lib/net.js

```diff
@@ -846,12 +846,16 @@ function connect(self, address, port, addressType, localAddress, localPort) {
 }
 
 
-// Check that the port number is not NaN when coerced to a number,
-// is an integer and that it falls within the legal range of port numbers.
+// Check that port is undefined, a number or a string.
```

Comment:
I understand why you made the changes, but it seems odd that this function accepts `undefined` as a valid port, and also that sometimes it returns a boolean and other times throws.